minecraft/1.7/src/main/java/nova/core/wrapper/mc/forge/v17/NovaMinecraftPreloader.java

```diff
@@ -88,97 +90,118 @@ public NovaMinecraftPreloader() {
 	 */
+	@SuppressWarnings("unchecked")
 	public static String generateSoundJSON(AbstractResourcePack pack) {
-		JsonObject fakeSoundJSON = new JsonObject();
+		StringWriter sw = new StringWriter();
 
-		for (String domain : (Set<String>) pack.getResourceDomains()) {
+		try (JsonGenerator json = Json.createGenerator(sw);) {
+			json.writeStartObject();
+			for (String domain : (Set<String>) pack.getResourceDomains()) {
 
-			if (pack instanceof FileResourcePack) {
-				//For zip resource packs
-				try {
-					ZipFile zipFile = new ZipFile(pack.resourcePackFile);
-
-					if (zipFile.getEntry("assets/" + domain + "/sounds/") != null) {
-						Enumeration zipEntries = zipFile.entries();
-
-						while (zipEntries.hasMoreElements()) {
-							String zipPath = ((ZipEntry) zipEntries.nextElement()).getName();
-
-							String prefix = "assets/" + domain + "/sounds/";
-							if (zipPath.startsWith(prefix) && !zipPath.equals(prefix)) {
-								String soundName = zipPath.replaceFirst(prefix, "").replaceFirst("[.][^.]+$", "");
-								ZipEntry entry = zipFile.getEntry(zipPath);
-
-								if (!soundName.contains("/")) {
-									JsonObject sound = new JsonObject();
-									sound.addProperty("category", "ambient");
-									JsonArray sounds = new JsonArray();
-
-									if (entry.isDirectory()) {
-										//Sound Collection
-										Enumeration zipEntries2 = zipFile.entries();
-										while (zipEntries2.hasMoreElements()) {
-											String zipPath2 = ((ZipEntry) zipEntries2.nextElement()).getName();
-
-											if (zipPath2.startsWith(prefix + soundName + "/") && !zipFile.getEntry(zipPath2).isDirectory()) {
-												String randomSoundName = zipPath2.replaceFirst(prefix + soundName + "/", "");
-												sounds.add(new JsonPrimitive(soundName + "/" + randomSoundName.replaceFirst("[.][^.]+$", "")));
+				if (pack instanceof FileResourcePack) {
+					//For zip resource packs
+					try {
+						ZipFile zipFile = new ZipFile(pack.resourcePackFile);
+
+						if (getZipEntryForResourcePack((FileResourcePack) pack, "assets/" + domain + "/sounds/") != null) {
+							for (Enumeration<? extends ZipEntry> zipEntries = zipFile.entries(); zipEntries.hasMoreElements();) {
+								String zipPath = zipEntries.nextElement().getName();
+								String prefix = "assets/" + domain + "/sounds/";
+								if (zipPath.toLowerCase().startsWith(prefix.toLowerCase()) && !zipPath.equalsIgnoreCase(prefix)) {
+									String soundName = zipPath.replaceFirst(prefix, "").replaceFirst("\\.[^\\.]+$", "");
+									ZipEntry entry = zipFile.getEntry(zipPath);
+
+									if (!soundName.contains("/")) {
+										json.writeStartObject(soundName);
+										json.write("category", "ambient");
+										json.writeStartArray("sounds");
+										if (entry.isDirectory()) {
+											//Sound Collection
+											for (Enumeration<? extends ZipEntry> zipEntries2 = zipFile.entries(); zipEntries2.hasMoreElements();) {
```

Comment:
I realise it was messy before, however I still think it would be nice to clean up if you can.